Formula/nu.rb

```diff
@@ -15,9 +15,45 @@ class Nu < Formula
 
   depends_on "pcre"
 
+  uses_from_macos "llvm" => [:build, :test]
+  uses_from_macos "swift" => :build # For libdispatch on Linux.
+
+  on_linux do
+    depends_on "gnustep-make" => :build
+    depends_on "gnustep-base"
+    depends_on "libffi"
+    depends_on "libobjc2"
+  end
+
+  # Clang must be used on Linux because GCC Objective-C support is insufficient.
+  fails_with :gcc
+
   def install
     ENV.delete("SDKROOT") if MacOS.version < :sierra
     ENV["PREFIX"] = prefix
+    # Some gnustep-make scripts located in libexecA.
+    ENV.append_path "PATH", Formula["gnustep-make"].libexec unless OS.mac?
+
+    unless OS.mac?
+      # Help linker find libdispatch from swift on Linux.
+      # This is only used for the mininush temporary compiler and is not needed for nush.
+      ldflags = %W[
+        "-L#{Formula["swift"].libexec}/lib/swift/linux"
+        "-Wl,-rpath,#{Formula["swift"].libexec}/lib/swift/linux"
+      ]
+      inreplace "Makefile", "LDFLAGS = ", "LDFLAGS = #{ldflags.join(" ")}"
```

Comment:
We can't pass these `LDFLAGS` to `make`? Or use `#change_make_var!`?

Reply:
I missed this comment earlier.  Unfortunately the line I was replacing is the one which seems to reset the `LDFLAGS` so passing it in doesn't work. Further complicating matters, the way the `LDFLAGS` are constructed in the Makefile seems incompatible with `change_make_var`:
```
LDFLAGS =
LDFLAGS += $(FRAMEWORKS)
LDFLAGS += $(LIBS)
LDFLAGS += $(LIBDIRS)
LDFLAGS += $(FFI_LIB)
ifeq ($(SYSTEM), Darwin)
else
        #LDFLAGS += $(shell gnustep-config --base-libs)
ifneq ($(SYSTEM), SunOS)
        LDFLAGS += -Wl,--rpath -Wl,/usr/local/lib

#LDFLAGS += `gnustep-config --debug-flags`
LDFLAGS += -L/usr/local/lib
LDFLAGS += -ldispatch
LDFLAGS += -lgnustep-base
# LDFLAGS += -lgnustep-gui
LDFLAGS += -L/usr/lib/GNUstep
```
So as far as I can tell `inreplace` seems to be the only way for this to work.

Reply:
And as soon as I pasted that, I realized I had missed that  `LIBDIRS`  is **not** overridden in the Makefile.  I was able to pass in the flags just by setting it in env, which is much cleaner than the `inreplace`. I'll push this change now.

Reply:
If you want to use `LDFLAGS` just add the flags you want to `ENV.ldflags` then pass `"LDFLAGS=#{ENV.ldflags}"` to `make`.

Reply:
I tried that and it doesn't seem to work because the first line in that part of the Makefile is `LDFLAGS = `, which I believe has the effect of resetting the `LDFLAGS` even if I pass them in as an argument.  I think if they were just using `+=` for all the assignments it would work as you described and that's indeed what is done with `LIBDIRS`, hence why it works.